# Dataquest.io Guided Project 1 - Profitable App Profiles for the App Store and Google Play Markets
Description:
Sources:
https://www.kaggle.com/lava18/google-play-store-apps/home
https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home
Goal: finding application characteristics most likely to maximize revenue from application

Defining explore function. Will be useful to take a look at
source databases. 

In [1]:
import operator
#importing operator library to sort dictionaries

def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))


Importing source datasets and transforming them to list variables.

In [2]:
def sum_column(table, C):
    result = sum(row[C] for row in table)
    return result

In [3]:
apple_open=open('AppleStore.csv',encoding="utf8")
google_open=open('googleplaystore.csv',encoding="utf8")
import csv
apple_read=csv.reader(apple_open)
google_read=csv.reader(google_open)
google_list=list(google_read)
apple_list=list(apple_read)

Exploration of datasets using created explore_data function.

In [4]:
explore_data(google_list,0,2,True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


Number of rows: 10842
Number of columns: 13


Google Store application dataset offes below structure:

| Variable       | Explanation                             | 
| -------------- | --------------------------------------- |
| App            | unique identifier.Variable name. String |
| Category       | type of app. String                     |
| Rating         | App rating. Float from 0 to 10          |
| Reviews        | Number of reviews. Integer              |
| Size           | Size in Megabytes. String               |
| Installs       | Number of installs range. String.       |
| Type           | 'Free' or 'Paid'. String                |
| Price          | Price in dollars. 0 for free apps.      |
| Content Rating | Age limit to use app. String            |
| Genres         | Genre of app. String                    |
| Last updated   | Date of last update. String             |
| Current Ver    | current version. String                 |
| Android Ver    | Min Android ver to use this app. String | 


In [5]:
explore_data(apple_list,0,2,True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


Number of rows: 7198
Number of columns: 16


Apple Store application dataset offes below structure:

| Variable         | Explanation                             | 
| ---------------  | --------------------------------------- |
| id               | unique application number. Key. Integer |
| track_name       | name of application. String             |
| size_bytes       | size of app in bytes. Integer           |
| currency         | currency of app price. String           |
| price            | price of app. 0 for free apps. Float.   |
| rating_count_tot | total number of ratings. Integer        |
| rating_count_ver | aaa                                     |
| user_rating      | Rating of users. Float from 0 to 10.    |
| user_rating_ver  | aaa                                     |
| ver              | application version. Float.             |
| cont_rating      | Content Rating on 5 star scale. String  |
| prime_genre      | App Genre. String                       |
| sup_devices.num  | number of supported devices. Integer    |
| ipadSc_urls.num  | aaa                                     |
| lang.num         | number of supported languages. Integer  |
| vpp_lic          | aaa                                     |

II Data cleansing
a) Preparing function to detect outliers and incorrect values.

In [6]:
#Dataset - name of dataset
#variable - case-sensitive name of variable in dataset
def variable_desc(dataset,variable):
    header=dataset[0]
    index_temp=int(header.index(variable))
    dict_temp={}
    for temp in dataset[1:]:
        b=str(temp[index_temp])
        if b in dict_temp:
            dict_temp[b] += 1
        else:
            dict_temp[b] = 1

    sorted_dict = list(sorted(dict_temp.items(), key=operator.itemgetter(1), reverse=True))
    print('Variable analysed: ' + variable)
    print('Number of distinct values:', len(sorted_dict))
    for row in sorted_dict:
        print(row)

In [7]:
variable_desc(google_list,'Category')

Variable analysed: Category
Number of distinct values: 34
('FAMILY', 1972)
('GAME', 1144)
('TOOLS', 843)
('MEDICAL', 463)
('BUSINESS', 460)
('PRODUCTIVITY', 424)
('PERSONALIZATION', 392)
('COMMUNICATION', 387)
('SPORTS', 384)
('LIFESTYLE', 382)
('FINANCE', 366)
('HEALTH_AND_FITNESS', 341)
('PHOTOGRAPHY', 335)
('SOCIAL', 295)
('NEWS_AND_MAGAZINES', 283)
('SHOPPING', 260)
('TRAVEL_AND_LOCAL', 258)
('DATING', 234)
('BOOKS_AND_REFERENCE', 231)
('VIDEO_PLAYERS', 175)
('EDUCATION', 156)
('ENTERTAINMENT', 149)
('MAPS_AND_NAVIGATION', 137)
('FOOD_AND_DRINK', 127)
('HOUSE_AND_HOME', 88)
('AUTO_AND_VEHICLES', 85)
('LIBRARIES_AND_DEMO', 85)
('WEATHER', 82)
('ART_AND_DESIGN', 65)
('EVENTS', 64)
('COMICS', 60)
('PARENTING', 60)
('BEAUTY', 53)
('1.9', 1)


b) Function to find duplicates among primary key

In [8]:
print(google_list[10473])
del google_list[10473]
print(google_list[10473])

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']
['osmino Wi-Fi: free WiFi', 'TOOLS', '4.2', '134203', '4.1M', '10,000,000+', 'Free', '0', 'Everyone', 'Tools', 'August 7, 2018', '6.06.14', '4.4 and up']


In [9]:
#Dataset - name of dataset
#variable - case-sensitive name of primary key variable in dataset
#output dataset - clean list based on primary key
#max_var - variable that we will use to deduplicate (we will take
# only max values)

def duplicates(dataset,primary_key,max_var):
    print('Length of input dataset - ' + str(len(dataset)))
    header=dataset[0]
    index_temp=int(header.index(primary_key))
    print('Index of primary key - ' + primary_key + ' : ' + str(index_temp))
    index_max=int(header.index(max_var))
    print('Index of variable used for deduplication - ' + max_var + ' : ' + str(index_max))
    dict_temp={}  
    clean_list=[]
    clean_list.append(header)
    for temp in dataset[1:]:   
        b=str(temp[index_temp])
        c=float(temp[index_max])
        if b in dict_temp and dict_temp[b] < c:
            dict_temp[b] = c
        if b not in dict_temp:
            dict_temp[b] = c
    print('Length of dictionary: ' + str(len(dict_temp)))
    deduped_list=[]
    already_added=[]
    deduped_list.append(header)
    for tempo in dataset[1:]:
        name=str(tempo[index_temp])
        d=float(tempo[index_max])
        if name not in already_added and d==dict_temp[name]:
            deduped_list.append(tempo)
            already_added.append(name)
    print('Length of deduplicated end-dataset: ' + str(len(deduped_list)))
    return deduped_list
    

In [10]:
google_dedup=duplicates(google_list,'App','Reviews')

Length of input dataset - 10841
Index of primary key - App : 0
Index of variable used for deduplication - Reviews : 3
Length of dictionary: 9659
Length of deduplicated end-dataset: 9660


In [11]:
#Source dataset
for a in google_list:
    if a[0]=='Microsoft Outlook':
        print(a)

['Microsoft Outlook', 'PRODUCTIVITY', '4.3', '3252896', '50M', '100,000,000+', 'Free', '0', 'Everyone', 'Productivity', 'August 2, 2018', '2.2.194', '4.4 and up']
['Microsoft Outlook', 'PRODUCTIVITY', '4.3', '3249518', '50M', '100,000,000+', 'Free', '0', 'Everyone', 'Productivity', 'August 2, 2018', '2.2.194', '4.4 and up']


In [12]:
#Dataset cleaned from app name duplicates
for a in google_dedup:
    if a[0]=='Microsoft Outlook':
        print(a)

['Microsoft Outlook', 'PRODUCTIVITY', '4.3', '3252896', '50M', '100,000,000+', 'Free', '0', 'Everyone', 'Productivity', 'August 2, 2018', '2.2.194', '4.4 and up']


In [13]:
apple_dedup=duplicates(apple_list,'id','rating_count_tot')
#No duplicates in apple_list dataset

Length of input dataset - 7198
Index of primary key - id : 0
Index of variable used for deduplication - rating_count_tot : 5
Length of dictionary: 7197
Length of deduplicated end-dataset: 7198


In [14]:
#Noneng - function returning list of english names from list potentially containing non-english names
#Recognition of non-english name - more than 3 non-ASCII characters 
#Dataset - name of dataset
#primary_key - case-sensitive name name containing non-english entries
#prnt - if prnt=1 then function prints all non-eng app names
def noneng(dataset,primary_key,prnt=0):
    print('Records in input dataset - ' + str(len(dataset)))
    header=dataset[0]
    index_temp=int(header.index(primary_key))
    print('Index of primary key - ' + primary_key + ' : ' + str(index_temp))
    list_noneng=[]
    clean_list=[]
    clean_list.append(header)
    for a in dataset[1:]:
        b=str(a[index_temp])
        c=0    
        for character in b:
            if ord(character)>127 and b not in list_noneng:
                c += 1
                if c>3 and b not in list_noneng:
                    list_noneng.append(b)
        if b not in list_noneng:
            clean_list.append(a)
    print('Number of records in english-app end-dataset: ' + str(len(clean_list)))
    if prnt==1:
        print('List of non-english apps')
        print(str(list_noneng))
    return clean_list         

In [15]:
google_noneng=noneng(google_dedup,'App',1)

Records in input dataset - 9660
Index of primary key - App : 0
Number of records in english-app end-dataset: 9615
List of non-english apps
['Flame - درب عقلك يوميا', 'သိင်္ Astrology - Min Thein Kha BayDin', 'РИА Новости', 'صور حرف H', 'L.POINT - 엘포인트 [ 포인트, 멤버십, 적립, 사용, 모바일 카드, 쿠폰, 롯데]', 'RMEduS - 음성인식을 활용한 R 프로그래밍 실습 시스템', 'AJ렌터카 법인 카셰어링', 'Al Quran Free - القرآن (Islam)', '中国語 AQリスニング', '日本AV历史', 'Ay Yıldız Duvar Kağıtları', 'বাংলা টিভি প্রো BD Bangla TV', 'Cъновник BG', 'CSCS BG (в български)', '뽕티비 - 개인방송, 인터넷방송, BJ방송', 'BL 女性向け恋愛ゲーム◆俺プリクロス', 'SecondSecret ‐「恋を読む」BLノベルゲーム‐', 'BL 女性向け恋愛ゲーム◆ごくメン', 'あなカレ【BL】無料ゲーム', '감성학원 BL 첫사랑', 'BQ-መጽሐፍ ቅዱሳዊ ጥያቄዎች', 'BS Calendar / Patro / पात्रो', 'Vip视频免费看-BT磁力搜索', 'Билеты ПДД CD 2019 PRO', 'Offline Jízdní řády CG Transit', 'Bonjour 2017 Abidjan CI ❤❤❤❤❤', 'CK 初一 十五', 'الفاتحون Conquerors', 'DG ग्राम / Digital Gram Panchayat', 'DM הפקות', 'DW فارسی By dw-arab.com', 'لعبة تقدر تربح DZ', 'বাংলাflix', 'RPG ブレイジング ソウルズ アクセレイト', '英漢字典 EC Dictionary', '

In [16]:
apple_noneng=noneng(apple_dedup,'track_name')

Records in input dataset - 7198
Index of primary key - track_name : 1
Number of records in english-app end-dataset: 6184


In [17]:
variable_desc(google_noneng,'Price')

Variable analysed: Price
Number of distinct values: 92
('0', 8864)
('$0.99', 145)
('$2.99', 124)
('$1.99', 73)
('$4.99', 70)
('$3.99', 56)
('$1.49', 45)
('$5.99', 26)
('$2.49', 25)
('$9.99', 19)
('$399.99', 11)
('$6.99', 10)
('$14.99', 9)
('$4.49', 9)
('$7.99', 7)
('$3.49', 7)
('$5.49', 5)
('$29.99', 5)
('$19.99', 5)
('$8.99', 5)
('$6.49', 5)
('$11.99', 3)
('$1.00', 3)
('$24.99', 3)
('$2.00', 3)
('$12.99', 3)
('$7.49', 2)
('$10.00', 2)
('$16.99', 2)
('$39.99', 2)
('$1.70', 2)
('$17.99', 2)
('$13.99', 2)
('$8.49', 2)
('$9.00', 1)
('$79.99', 1)
('$10.99', 1)
('$1.50', 1)
('$15.99', 1)
('$33.99', 1)
('$74.99', 1)
('$3.95', 1)
('$3.88', 1)
('$25.99', 1)
('$400.00', 1)
('$3.02', 1)
('$1.76', 1)
('$4.84', 1)
('$4.77', 1)
('$1.61', 1)
('$2.50', 1)
('$1.59', 1)
('$1.29', 1)
('$5.00', 1)
('$299.99', 1)
('$379.99', 1)
('$37.99', 1)
('$18.99', 1)
('$389.99', 1)
('$19.90', 1)
('$1.75', 1)
('$14.00', 1)
('$4.85', 1)
('$46.99', 1)
('$109.99', 1)
('$154.99', 1)
('$3.08', 1)
('$2.59', 1)
('$4.80', 1)


In [18]:
variable_desc(apple_noneng,'price')

Variable analysed: price
Number of distinct values: 34
('0.0', 3222)
('2.99', 669)
('0.99', 641)
('1.99', 610)
('4.99', 375)
('3.99', 266)
('6.99', 165)
('9.99', 76)
('5.99', 43)
('7.99', 30)
('14.99', 15)
('19.99', 13)
('8.99', 8)
('24.99', 8)
('13.99', 6)
('29.99', 6)
('15.99', 4)
('17.99', 3)
('11.99', 3)
('59.99', 3)
('39.99', 2)
('16.99', 2)
('49.99', 2)
('20.99', 1)
('12.99', 1)
('74.99', 1)
('249.99', 1)
('27.99', 1)
('22.99', 1)
('18.99', 1)
('99.99', 1)
('21.99', 1)
('34.99', 1)
('299.99', 1)


In [19]:
#free - function returning free applications from a super-set of paid and free applications
#Recognition of non-english name - more than 3 non-ASCII characters 
#Dataset - name of dataset
#primary_key - case-sensitive name name containing non-english entries
#prnt - if prnt=1 then function prints all non-eng app names
def free(dataset,primary_key):
    print('Records in input dataset - ' + str(len(dataset)))
    header=dataset[0]
    index_temp=int(header.index(primary_key))
    print('Index of primary key - ' + primary_key + ' : ' + str(index_temp))
    paid_apps=[]
    free_apps=[]
    free_apps.append(header)
    for a in dataset[1:]:
        b=str(a[index_temp])  
        if b != '0' and b != '0.0':
            paid_apps.append(b)
        else:
            free_apps.append(a)
    print('Number of records in free app end-dataset: ' + str(len(free_apps)))
    return free_apps    

In [20]:
free_google=free(google_noneng,'Price')

Records in input dataset - 9615
Index of primary key - Price : 7
Number of records in free app end-dataset: 8865


In [21]:
free_apple=free(apple_noneng,'price')

Records in input dataset - 6184
Index of primary key - price : 4
Number of records in free app end-dataset: 3223


#Build a minimal Android version of the app, and add it to Google Play.
If the app has a good response from users, we then develop it further.
If the app is profitable after six months, we also build an iOS version of the app and add it to the App Store.
Variables we are going to inspect:

*Google dataset
*Rating
*Size
*Content Rating
*Genre

In [22]:
#Dataset - name of dataset
#variable - case-sensitive name of variable in dataset
def freq(dataset,variable):
    header=dataset[0]
    index_temp=int(header.index(variable))
    dict_temp={}
    for temp in dataset[1:]:
        b=str(temp[index_temp])
        if b in dict_temp:
            dict_temp[b] += 1
        else:
            dict_temp[b] = 1

    sorted_dict = list(sorted(dict_temp.items(), key=operator.itemgetter(1), reverse=True))
    total=sum_column(sorted_dict,1)  
    print('Variable analysed: ' + variable)
    print('Number of distinct values:', len(sorted_dict))
    print('Total observations: ', total)
    for row in sorted_dict:
        print(row, ", Fraction of total:", round(row[1]/total*100,1),'%')

In [23]:
freq(free_apple,'prime_genre')

Variable analysed: prime_genre
Number of distinct values: 23
Total observations:  3222
('Games', 1874) , Fraction of total: 58.2 %
('Entertainment', 254) , Fraction of total: 7.9 %
('Photo & Video', 160) , Fraction of total: 5.0 %
('Education', 118) , Fraction of total: 3.7 %
('Social Networking', 106) , Fraction of total: 3.3 %
('Shopping', 84) , Fraction of total: 2.6 %
('Utilities', 81) , Fraction of total: 2.5 %
('Sports', 69) , Fraction of total: 2.1 %
('Music', 66) , Fraction of total: 2.0 %
('Health & Fitness', 65) , Fraction of total: 2.0 %
('Productivity', 56) , Fraction of total: 1.7 %
('Lifestyle', 51) , Fraction of total: 1.6 %
('News', 43) , Fraction of total: 1.3 %
('Travel', 40) , Fraction of total: 1.2 %
('Finance', 36) , Fraction of total: 1.1 %
('Weather', 28) , Fraction of total: 0.9 %
('Food & Drink', 26) , Fraction of total: 0.8 %
('Reference', 18) , Fraction of total: 0.6 %
('Business', 17) , Fraction of total: 0.5 %
('Book', 14) , Fraction of total: 0.4 %
('Navig

In [24]:
freq(free_google,'Category')

Variable analysed: Category
Number of distinct values: 33
Total observations:  8864
('FAMILY', 1676) , Fraction of total: 18.9 %
('GAME', 862) , Fraction of total: 9.7 %
('TOOLS', 750) , Fraction of total: 8.5 %
('BUSINESS', 407) , Fraction of total: 4.6 %
('LIFESTYLE', 346) , Fraction of total: 3.9 %
('PRODUCTIVITY', 345) , Fraction of total: 3.9 %
('FINANCE', 328) , Fraction of total: 3.7 %
('MEDICAL', 313) , Fraction of total: 3.5 %
('SPORTS', 301) , Fraction of total: 3.4 %
('PERSONALIZATION', 294) , Fraction of total: 3.3 %
('COMMUNICATION', 287) , Fraction of total: 3.2 %
('HEALTH_AND_FITNESS', 273) , Fraction of total: 3.1 %
('PHOTOGRAPHY', 261) , Fraction of total: 2.9 %
('NEWS_AND_MAGAZINES', 248) , Fraction of total: 2.8 %
('SOCIAL', 236) , Fraction of total: 2.7 %
('TRAVEL_AND_LOCAL', 207) , Fraction of total: 2.3 %
('SHOPPING', 199) , Fraction of total: 2.2 %
('BOOKS_AND_REFERENCE', 190) , Fraction of total: 2.1 %
('DATING', 165) , Fraction of total: 1.9 %
('VIDEO_PLAYERS',

In [25]:
freq(free_google,'Genres')

Variable analysed: Genres
Number of distinct values: 114
Total observations:  8864
('Tools', 749) , Fraction of total: 8.4 %
('Entertainment', 538) , Fraction of total: 6.1 %
('Education', 474) , Fraction of total: 5.3 %
('Business', 407) , Fraction of total: 4.6 %
('Lifestyle', 345) , Fraction of total: 3.9 %
('Productivity', 345) , Fraction of total: 3.9 %
('Finance', 328) , Fraction of total: 3.7 %
('Medical', 313) , Fraction of total: 3.5 %
('Sports', 307) , Fraction of total: 3.5 %
('Personalization', 294) , Fraction of total: 3.3 %
('Communication', 287) , Fraction of total: 3.2 %
('Action', 275) , Fraction of total: 3.1 %
('Health & Fitness', 273) , Fraction of total: 3.1 %
('Photography', 261) , Fraction of total: 2.9 %
('News & Magazines', 248) , Fraction of total: 2.8 %
('Social', 236) , Fraction of total: 2.7 %
('Travel & Local', 206) , Fraction of total: 2.3 %
('Shopping', 199) , Fraction of total: 2.2 %
('Books & Reference', 190) , Fraction of total: 2.1 %
('Simulation', 1

In [26]:
#searches for maximum value of variables that can be converted to float type
def max_value(dataset,variable):
    header=dataset[0]
    index_temp=int(header.index(variable))
    values_list = [float(e[index_temp]) for e in dataset[1:]]
    result = max(values_list)
    print('Maximum value for ' + str(variable) + ' : ' + str(result))

In [28]:
max_value(free_google,'Reviews')
#na reviews dziala dobrze ale na skategoryzowanych nie, 
#czy mozna to lepiej zrobic

Maximum value for Reviews : 78158306.0


In [46]:
#WHERE function - returns rows equal to 'where' condition
def print_where(dataset,variable,condition):
    header=dataset[0]
    index_temp=int(header.index(variable))
    for a in dataset:
        if a[index_temp]==condition:
            print(a)

In [48]:
print_where(free_google,'Reviews','78158306')

['Facebook', 'SOCIAL', '4.1', '78158306', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'August 3, 2018', 'Varies with device', 'Varies with device']


In [50]:
freq(free_google,'Installs')

Variable analysed: Installs
Number of distinct values: 21
Total observations:  8864
('1,000,000+', 1394) , Fraction of total: 15.7 %
('100,000+', 1024) , Fraction of total: 11.6 %
('10,000,000+', 935) , Fraction of total: 10.5 %
('10,000+', 904) , Fraction of total: 10.2 %
('1,000+', 744) , Fraction of total: 8.4 %
('100+', 613) , Fraction of total: 6.9 %
('5,000,000+', 605) , Fraction of total: 6.8 %
('500,000+', 493) , Fraction of total: 5.6 %
('50,000+', 423) , Fraction of total: 4.8 %
('5,000+', 400) , Fraction of total: 4.5 %
('10+', 314) , Fraction of total: 3.5 %
('500+', 288) , Fraction of total: 3.2 %
('50,000,000+', 204) , Fraction of total: 2.3 %
('100,000,000+', 189) , Fraction of total: 2.1 %
('50+', 170) , Fraction of total: 1.9 %
('5+', 70) , Fraction of total: 0.8 %
('1+', 45) , Fraction of total: 0.5 %
('500,000,000+', 24) , Fraction of total: 0.3 %
('1,000,000,000+', 20) , Fraction of total: 0.2 %
('0+', 4) , Fraction of total: 0.0 %
('0', 1) , Fraction of total: 0.0 

In [51]:
print_where(free_google,'Installs','1,000,000,000+')

['Google Play Books', 'BOOKS_AND_REFERENCE', '3.9', '1433233', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Books & Reference', 'August 3, 2018', 'Varies with device', 'Varies with device']
['WhatsApp Messenger', 'COMMUNICATION', '4.4', '69119316', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Everyone', 'Communication', 'August 3, 2018', 'Varies with device', 'Varies with device']
['Messenger – Text and Video Chat for Free', 'COMMUNICATION', '4.0', '56646578', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Everyone', 'Communication', 'August 1, 2018', 'Varies with device', 'Varies with device']
['Skype - free IM & video calls', 'COMMUNICATION', '4.1', '10484169', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Everyone', 'Communication', 'August 3, 2018', 'Varies with device', 'Varies with device']
['Google Chrome: Fast & Secure', 'COMMUNICATION', '4.3', '9643041', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Everyone', 'Communication', '

In [31]:
max_value(free_apple,'rating_count_tot')

Maximum value for rating_count_tot : 2974676.0


In [54]:
print_where(free_apple,'rating_count_tot','2974676')

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


In [55]:
max_value(free_apple,'user_rating')

Maximum value for user_rating : 5.0


In [56]:
print_where(free_apple,'user_rating','5.0')

['487119327', 'Head Soccer', '121319424', 'USD', '0.0', '481564', '8518', '5.0', '5.0', '5.4.1', '4+', 'Games', '40', '4', '11', '1']
['930574573', 'Sniper 3D Assassin: Shoot to Kill Gun Game', '157851648', 'USD', '0.0', '386521', '10332', '5.0', '5.0', '1.17.6', '17+', 'Games', '40', '5', '9', '1']
['698255242', 'Geometry Dash Lite', '69076992', 'USD', '0.0', '370370', '2650', '5.0', '4.5', '2.11', '4+', 'Games', '37', '4', '1', '1']
['436491861', "Domino's Pizza USA", '105743360', 'USD', '0.0', '258624', '2481', '5.0', '4.5', '4.2.0', '4+', 'Food & Drink', '37', '5', '2', '1']
['887947640', 'CSR Racing 2', '1944321024', 'USD', '0.0', '257100', '2025', '5.0', '5.0', '1.11.3', '4+', 'Games', '37', '5', '11', '1']
['600674056', "Pictoword: Fun 2 Pics Guess What's the Word Trivia", '126216192', 'USD', '0.0', '186089', '1010', '5.0', '4.5', '2.4.2', '4+', 'Games', '37', '5', '1', '1']
['606190854', 'Iron Force', '124666880', 'USD', '0.0', '141634', '1655', '5.0', '4.0', '2.4.1', '17+', 'G

# Opisz wyniki i jaka aplikacje powinnismy zrobic,
1. 'Games' is the most dominant free Apple prime genre with 1874 applications, 'Entertainment' is second with 254 applications
2. 'Family' is the most dominant free Google category with 1676 applications, 'Game' is second with 862 applications
3. 'Tools' is the most dominant free Google genre with 749 applications, 'Entertainment' is second with 538 applications
4. Most of applications are designed for entertainment purposes - if we want to create a succesfull free application based on genre and category a game that can be enjoyed by whole family would be the right choice
5. We have to count as well for applications that have the most users - lets check that.